dbt_platform_helper/commands/conduit.py

```diff
@@ -222,15 +223,29 @@ def create_addon_client_task(
     elif addon_type == "redis" or addon_type == "opensearch":
         secret_name += "_ENDPOINT"
 
-    subprocess.call(
-        f"copilot task run --app {app.name} --env {env} "
-        f"--task-group-name {task_name} "
-        f"--image {CONDUIT_DOCKER_IMAGE_LOCATION}:{addon_type} "
-        f"--secrets CONNECTION_SECRET={get_connection_secret_arn(app, env, secret_name)} "
-        "--platform-os linux "
-        "--platform-arch arm64",
-        shell=True,
-    )
+    try:
+        session.client("iam").get_role(RoleName=f"{app.name}-{addon_type}-{app.name}-{env}-ecsTask")
+
+        subprocess.call(
+            f"copilot task run --app {app.name} --env {env} "
+            f"--task-group-name {task_name} "
+            f"--execution-role {app.name}-{addon_type}-{app.name}-{env}-ecsTask "
+            f"--image {CONDUIT_DOCKER_IMAGE_LOCATION}:{addon_type} "
+            f"--secrets CONNECTION_SECRET={get_connection_secret_arn(app, env, secret_name)} "
+            "--platform-os linux "
+            "--platform-arch arm64",
+            shell=True,
+        )
+    except:
+        subprocess.call(
+            f"copilot task run --app {app.name} --env {env} "
+            f"--task-group-name {task_name} "
+            f"--image {CONDUIT_DOCKER_IMAGE_LOCATION}:{addon_type} "
+            f"--secrets CONNECTION_SECRET={get_connection_secret_arn(app, env, secret_name)} "
+            "--platform-os linux "
+            "--platform-arch arm64",
+            shell=True,
+        )
```

Comment:
Under what circumstance would we want to make the call without specifying the execution role?

Reply:
@WillGibson If/when the IAM role created by Terraform doesn't exist yet.

Reply:
Isn't it already broken in that case? Or am I missing something?

Reply:
@mramplin so that would make my comment a bit more useful? if the role hasn't been created then the IAM call would be the one raising the exception, correct?

also, probably worth making the `except` clause a bit more specific and have it catch the correct one

Reply:
@WillGibson it's already broken for Redis, but currently OpenSearch still works without that switch. I have also added the logic into the Terraform to enable the CMK with OpenSearch; so it will work with the execution role switch. Can we assume that the Terraform will be up to date? If we can then I can remove the check.

Reply:
> Can we assume that the Terraform will be up to date?

No, don't remove it.

Thanks for explaining @mramplin.

I think @samuele-mattiuzzo has something with the duplication reduction idea though.

Reply:
Updates suggested by @samuele-mattiuzzo  have been made and regression [tests reran](https://eu-west-2.console.aws.amazon.com/codesuite/codebuild/763451185160/projects/platform-tools-test?region=eu-west-2&builds-meta=eyJmIjp7InRleHQiOiIifSwicyI6e30sIm4iOjIwLCJpIjowfQ#:~:text=platform%2Dtools%2Dtest%3Ab1e66874%2Df172%2D4ff9%2Dad4e%2D138bdcb7d858)

Thanks both for the feedback.